test/semantics/allocate11.f90

```diff
@@ -0,0 +1,185 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! Check for semantic errors in ALLOCATE statements
+
+! TODO: Function Pointer in allocate and derived types!
+
+! Rules I should know when working wit coarrays and derived type:
+
+! C736: If EXTENDS appears and the type being defined has a coarray ultimate
+! component, its parent type shall have a coarray ultimate component.
+
+! C746: (R737) If a coarray-spec appears, it shall be a deferred-coshape-spec-list
+! and the component shall have the ALLOCATABLE attribute.
+
+! C747: If a coarray-spec appears, the component shall not be of type C_PTR or
+! C_FUNPTR from the intrinsic module ISO_C_BINDING (18.2), or of type TEAM_TYPE from the
+! intrinsic module ISO_FORTRAN_ENV (16.10.2).
+
+! C748: A data component whose type has a coarray ultimate component shall be a
+! nonpointer nonallocatable scalar and shall not be a coarray.
+
+! 7.5.4.3 Coarray components
+! 7.5.6 Final subroutines: C786
+
+
+! C825 An entity whose type has a coarray ultimate component shall be a
+! nonpointer nonallocatable scalar, shall not be a coarray, and shall not be a function result.
+
+! C826 A coarray or an object with a coarray ultimate component shall be an
+! associate name, a dummy argument, or have the ALLOCATABLE or SAVE attribute.
+
+subroutine C937(var)
+! Type-spec shall not specify a type that has a coarray ultimate component.
+
+
+  type A
+    real, allocatable :: x[:]
+  end type
+
+  type B
+    type(A) y
+    type(B), pointer :: forward
+    real :: u
+  end type
+
+  type C
+    type(B) z
+  end type
+
+  type D
+    type(A), pointer :: potential
+  end type
+
+
+
+  class(*), allocatable :: var
+  ! unlimited polymorphic is the ONLY way to get an allocatable/pointer 'var' that can be
+  ! allocated with a type-spec T that has coarray ultimate component.
+  ! Rational:
```

Comment:
Spelling: "Rationale"

Reply:
Fixed. Also rephrased part of the comment that were wrong.